src/platform/plugins/shared/dashboard/public/dashboard_app/listing_page/dashboard_listing_page.test.tsx

```diff
@@ -41,95 +41,71 @@ jest.mock('../no_data/dashboard_app_no_data', () => {
   dataService.dataViews.hasData.hasUserDataView = jest.fn().mockResolvedValue(false);
 
-  let component: ReactWrapper;
-  await act(async () => {
-    ({ component } = mountWith({}));
+  renderDashboardListingPage();
+
+  await waitFor(() => {
+    expect(DashboardAppNoDataPage).toHaveBeenCalled();
   });
-  component!.update();
-  expect(DashboardAppNoDataPage).toHaveBeenCalled();
 });
 
 test('initialFilter is passed through if title is not provided', async () => {
-  const props = makeDefaultProps();
-  props.initialFilter = 'filterPassThrough';
+  const initialFilter = faker.lorem.word();
 
-  let component: ReactWrapper;
+  renderDashboardListingPage({ initialFilter });
 
-  await act(async () => {
-    ({ component } = mountWith({ props }));
+  await waitFor(() => {
+    expect(DashboardListing).toHaveBeenCalledWith(
+      expect.objectContaining({ initialFilter }),
+      expect.any(Object) // react context
+    );
   });
-
-  component!.update();
-  expect(DashboardListing).toHaveBeenCalledWith(
-    expect.objectContaining({ initialFilter: 'filterPassThrough' }),
-    expect.any(Object) // react context
-  );
 });
 
 test('When given a title that matches multiple dashboards, filter on the title', async () => {
-  const title = 'search by title';
-  const props = makeDefaultProps();
-  props.title = title;
-
   (dashboardContentManagementService.findDashboards.findByTitle as jest.Mock).mockResolvedValue(
     undefined
   );
 
-  let component: ReactWrapper;
+  const redirectTo = jest.fn();
 
-  await act(async () => {
-    ({ component } = mountWith({ props }));
-  });
-  component!.update();
+  renderDashboardListingPage({ title: 'search by title', redirectTo });
 
-  expect(props.redirectTo).not.toHaveBeenCalled();
-  expect(DashboardListing).toHaveBeenCalledWith(
-    expect.objectContaining({ initialFilter: 'search by title' }),
-    expect.any(Object) // react context
-  );
+  await waitFor(() => {
+    expect(redirectTo).not.toHaveBeenCalled();
+    expect(DashboardListing).toHaveBeenCalledWith(
+      expect.objectContaining({ initialFilter: 'search by title' }),
+      expect.any(Object) // react context
+    );
+  });
 });
 
 test('When given a title that matches one dashboard, redirect to dashboard', async () => {
-  const title = 'search by title';
-  const props = makeDefaultProps();
-  props.title = title;
   (dashboardContentManagementService.findDashboards.findByTitle as jest.Mock).mockResolvedValue({
     id: 'you_found_me',
   });
+  const redirectTo = jest.fn();
 
-  let component: ReactWrapper;
+  renderDashboardListingPage({ title: 'search by title', redirectTo });
 
-  await act(async () => {
-    ({ component } = mountWith({ props }));
-  });
-  component!.update();
-  expect(props.redirectTo).toHaveBeenCalledWith({
-    destination: 'dashboard',
-    id: 'you_found_me',
-    useReplace: true,
+  await waitFor(() => {
+    expect(redirectTo).toHaveBeenCalledWith({
```

Comment:
I think it might be in our interest to _remove_ this test and the functionality it covers.

For some reason we have a hidden / legacy capability that allows the Dashboard Listing page to redirect to a Dashboard by title when given a title in the URL. I re-wrote that functionality a few times because I didn't want to remove it, but now that I think about it more it's just cruft and I am very confident no one is using this.

Reply:
Could we submit a separate issue for it? That was a very quick PR and I don't want to add more 'side' tasks to emotion rewrite (which is what triggered me doing this rewrite) 😅